### PR TITLE
STOR-1402: Chore: update libraries in all operators

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/STOR-1402

This bumps all libraries for OCP 4.15 / k8s 1.28.
controller-runtime removed some deprecated fields that LSO was still using,
which required some changes in main, diskmaker_manager, and unit tests.

cc @openshift/storage

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -99,10 +101,11 @@ func main() {
 	le := utils.GetLeaderElectionConfig(restConfig, enableLeaderElection)
 
 	mgr, err := ctrl.NewManager(restConfig, ctrl.Options{
-		Namespace:              namespace,
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{namespace: {}},
+		},
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
```
Comment (dobsonj) [inline main.go:105]: This doesn't need to be set, the new API defaults to 9443 already.
https://github.com/openshift/local-storage-operator/blob/e2eec8b9380c62a2eebeb0b893ea59350e3f6914/vendor/sigs.k8s.io/controller-runtime/pkg/webhook/server.go#L76-L78

File: go.mod [hunk 2]
```diff
@@ -73,25 +73,26 @@ require (
...
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/apiserver v0.27.4 // indirect
-	k8s.io/component-base v0.27.4 // indirect
-	k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f // indirect
+	k8s.io/apiserver v0.28.2 // indirect
+	k8s.io/component-base v0.28.2 // indirect
+	k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 // indirect
 	k8s.io/kubernetes v1.25.1 // indirect
 	k8s.io/mount-utils v0.25.1 // indirect
```
Comment (jsafrane) [inline go.mod:97]: bump these too. They're replaced below anyway.

**Review by jsafrane — COMMENTED** [review]
I like cleaning up test variable names!

File: go.mod (hunk 2 above)
Comment (dobsonj) [reply go.mod]: ack, updated.

**Comment by dobsonj** [comment]
CI should be enough for this library bump, and we have other changes that depend on it. No user-facing changes.
/label docs-approved
/label px-approved
/label qe-approved

**Pull request merged**